client/src/components/auth/loginForm.tsx

```diff
@@ -1,90 +1,102 @@
-          <p>App explanation</p>
+
+    async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+        e.preventDefault();
+
+        let newUrl: LoginOutput = await handleLogin(userInfo);
+
+        localStorage.setItem("userId", newUrl.id);
+        navigate(`/app/dashboard/${newUrl.id}`);
+    }
+
+    
+    return (
+      <>
+        <header className="row-items">
+          <h1>Memoriez</h1>
+          <a href="/">Register</a>
+        </header>
+
+        <div className="split left">
+          <div className="centered">
+            <h2>Other component</h2>
+            <p>App explanation</p>
+          </div>
         </div>
-      </div>
 
-      <div className="split right">
-        <div className="centered">
-          <main>
-            <section>
-              <h1>Login here!</h1>
-              <form
-                onSubmit={async (e) => {
-                  handleSubmit(e);
-                }}
-              >
-                <div className="input">
-                  <label htmlFor="username">Username:</label>
-                  <input
-                    id="username"
-                    type={"text"}
-                    onChange={(e) => setUsername(e.target.value)}
-                  ></input>
-                </div>
-                <div className="input">
-                  <label htmlFor="mail">Mail:</label>
-                  <input
-                    id="mail"
-                    type={"text"}
-                    onChange={(e) => setMail(e.target.value)}
-                  ></input>
-                </div>
-                <div className="simple-center">
-                  <input
-                    type={"submit"}
-                    value={"Login"}
-                    className="button"
-                  ></input>
-                </div>
-              </form>
-            </section>
-          </main>
+        <div className="split right">
+          <div className="centered">
+            <main>
+              <section>
+                <h1>Login here!</h1>
+                <form
+                  onSubmit={async (e: React.FormEvent<HTMLFormElement>) => {
+                    handleSubmit(e);
+                  }}
+                >
+                    <div className="input">
+                        <label htmlFor="username">Username:</label>
+                        <input
+                            id="username"
+                            type={"text"}
+                            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                            setUsername(e.target.value)
+                            }
+                        ></input>
+                    </div>
+
+                    <div className="input">
+                        <label htmlFor="mail">Mail:</label>
+                        <input
+                            id="mail"
+                            type={"text"}
+                            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
```

Comment:
No need to type it: this is not something you create (onChange is by React), so you can rely on their typing